packages/prepare.js

```diff
@@ -0,0 +1,47 @@
+const fs = require('fs').promises;
+const path = require('path');
+const util = require('util');
+const execFile = util.promisify(require('child_process').execFile);
+
+const rootDir = path.resolve(__dirname, '..');
+const srcDir = path.join(rootDir, 'ed/idl');
+const patchDir = path.join(rootDir, 'ed/idlpatches');
+const dstDir = path.join(rootDir, 'packages/idl');
+
+async function main() {
+  // rm dstDir/*.idl
+  const dstFiles = await fs.readdir(dstDir);
+  for (const file of dstFiles) {
+    if (file.endsWith('.idl')) {
+      await fs.unlink(path.join(dstDir, file));
+    }
+  }
+
+  // cp srcDir/*.idl dstDir/
+  const srcFiles = await fs.readdir(srcDir);
+  for (const file of srcFiles) {
+    if (file.endsWith('.idl')) {
+      await fs.copyFile(path.join(srcDir, file), path.join(dstDir, file));
+    }
+  }
+
+  // The patches are against ed/idl/ and can be applied there using `git am`,
+  // but apply them in packages/idl/ instead using `git apply`.
+  // (The --binary option disables the heuristic for transforming CRLF line
+  // endings into LF line endings on Windows machines)
+  const patchFiles = await fs.readdir(patchDir);
+  for (const file of patchFiles) {
+    if (file.endsWith('.patch')) {
+      const patch = path.join(patchDir, file);
+      console.log(`Applying ${file}`);
+      await execFile('git', ['apply', '--binary', '--directory=packages/idl', '-p3', patch], {
```

Comment:
The `--binary` option seems useless on Windows machines when `git am` is used.

Actually, looking at the [documentation for `git am`](https://git-scm.com/docs/git-apply/2.29.0#Documentation/git-apply.txt---binary), it is now just a no-op, so can be dropped (along with the comment that explains why it's there).

Reply:
Great, I've done that cleanup now.